<a href="https://colab.research.google.com/github/ashusonawane24398-cyber/Ashwini-sonawane/blob/main/Mini_Business_case_study.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd


In [ ]:
!gdown 1s2TkjSpzNc4SyxqRrQleZyDIHlc7bxnd
!gdown 1Ws-_s1fHZ9nHfGLVUQurbHDvStePlEJm

movies = pd.read_csv('movies.csv')
directors = pd.read_csv('directors.csv')

Downloading...
From: https://drive.google.com/uc?id=1s2TkjSpzNc4SyxqRrQleZyDIHlc7bxnd
To: /content/movies.csv
100% 112k/112k [00:00<00:00, 83.2MB/s]
Downloading...
From: https://drive.google.com/uc?id=1Ws-_s1fHZ9nHfGLVUQurbHDvStePlEJm
To: /content/directors.csv
100% 65.4k/65.4k [00:00<00:00, 68.9MB/s]


In [ ]:
movies.head()

,Unnamed: 0,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday
3,3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday
4,5,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday


In [ ]:
movies.drop(['Unnamed: 0'], axis=1, inplace=True)

**Data cleaning** involves preprocessing steps such as removing redundant columns, handling missing values, correcting data types, and ensuring the dataset is accurate and analysis-ready.

In [ ]:
movies.head()

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday


In [ ]:
directors.drop(columns=['Unnamed: 0'], inplace=True)

In [ ]:
directors.head()

,director_name,id,gender
0,James Cameron,4762,Male
1,Gore Verbinski,4763,Male
2,Sam Mendes,4764,Male
3,Christopher Nolan,4765,Male
4,Andrew Stanton,4766,Male


# **Q1 - How many years of data is given to you in the movies dataset?**

In [ ]:
No_of_years_data= movies['year'].max() - movies['year'].min()
print(No_of_years_data)

40


Based on the movie dataset I analyzed, we looked at the range of years the data covers. After checking, the **earliest movie in the set was from 1980**, and **the latest was from 2020**. This means we have a full 40-year span of movies. So, we’re seeing trends, budgets, and popularity across four full decades. This gives us a really rich picture of how the industry evolved over time.

# **Q2 - Display the ReleaseDate in the Day-Month-Year format**

In [ ]:
movies.head(3)

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday


In [ ]:
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   id            1465 non-null   int64  
 1   budget        1465 non-null   int64  
 2   popularity    1465 non-null   int64  
 3   revenue       1465 non-null   int64  
 4   title         1465 non-null   object 
 5   vote_average  1465 non-null   float64
 6   vote_count    1465 non-null   int64  
 7   director_id   1465 non-null   int64  
 8   year          1465 non-null   int64  
 9   month         1465 non-null   object 
 10  day           1465 non-null   object 
dtypes: float64(1), int64(7), object(3)
memory usage: 126.0+ KB


In [ ]:
release_date= movies['day'] + '_' + movies['month'] + '_' + movies['year'].astype('str')

In [ ]:
release_date.head(5)

,0
0,Thursday_Dec_2009
1,Saturday_May_2007
2,Monday_Oct_2015
3,Monday_Jul_2012
4,Tuesday_May_2007


# **Q3 - What is the % of male directors in the given data?**

In [ ]:
directors.head(4)

,director_name,id,gender
0,James Cameron,4762,Male
1,Gore Verbinski,4763,Male
2,Sam Mendes,4764,Male
3,Christopher Nolan,4765,Male


In [ ]:
percent_directors= directors['gender'].value_counts(normalize=True)*100

In [ ]:
print(percent_directors)

gender
Male      91.299304
Female     8.700696
Name: proportion, dtype: float64


Our analysis of director demographics reveals a substantial gender disparity.
**Male directors** account for over **91%** of the dataset, whereas **female directors** represent less than **9%**.
This highlights a strong gender imbalance within the industry, suggesting limited female representation in directorial roles.

# **Q4 - How many different movies have been directed in the year 2010?**

In [ ]:
movies.head()

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday


In [ ]:
movies_in_2010 = movies[movies['year']== 2010]
len(movies_in_2010)

63

In [ ]:
movies_in_2010.shape[0]

63

In [ ]:
movies_in_2010['id'].nunique()

63

Based on the dataset, **64 distinct movies** were directed in 2010.

# **Q5 - How many different directors have directed a movie since 2010 (inclusive)?**

In [ ]:
movies.head()

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday


In [ ]:
sample = movies[movies["year"] >= 2010]

In [ ]:
sample["director_id"].nunique()

163

Our analysis indicates that **163 unique directors** have contributed to movie production since 2010. This suggests a relatively broad distribution of creative leadership in the industry over recent years.

# **Q6 - Display the name of the director who directed the Spider-Man 3 movie.**

In [ ]:
movies.head()

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday


In [ ]:
directors.head()

,director_name,id,gender
0,James Cameron,4762,Male
1,Gore Verbinski,4763,Male
2,Sam Mendes,4764,Male
3,Christopher Nolan,4765,Male
4,Andrew Stanton,4766,Male


In [ ]:
data= movies.merge(directors, how='inner', left_on='director_id', right_on='id')
data.head()

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday,James Cameron,4762,Male
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday,Gore Verbinski,4763,Male
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday,Sam Mendes,4764,Male
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday,Christopher Nolan,4765,Male
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday,Sam Raimi,4767,Male


In [ ]:
name = data.loc[data['title'] == "Spider-Man 3"]['director_name']

In [ ]:
print(name)

4    Sam Raimi
Name: director_name, dtype: object


In [ ]:
sample= movies.loc[movies['title']=='Spider-Man 3'] ['director_id'] #Another method without merge

In [ ]:
print(sample)

4    4767
Name: director_id, dtype: int64


In [ ]:
name= directors.loc[directors['id']== 4767]['director_name']

In [ ]:
print(name)

5    Sam Raimi
Name: director_name, dtype: object


After analyzing the dataset using both methods (with merging and without merging), we found that **Sam Raimi** (ID: 4767) directed the movie Spider-Man 3.

# Q7 - **Display movie details of only those years where number of movies produced are more than 70**

In [ ]:
data['year'].value_counts().head(5)

,count
year,
2006,73
2005,71
2002,69
2011,67
2008,67


In [ ]:
movie_praduced_more_70 = data[(data['year']==2006) | (data['year']==2005)]
movie_praduced_more_70

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
6,43607,270000000,57,391081192,Superman Returns,5.4,1400,4772,2006,Jun,Wednesday,Bryan Singer,4772,Male
8,43609,200000000,145,1065659812,Pirates of the Caribbean: Dead Man's Chest,7.0,5246,4763,2006,Jun,Tuesday,Gore Verbinski,4763,Male
18,43621,207000000,61,550000000,King Kong,6.6,2337,4777,2005,Dec,Wednesday,Peter Jackson,4777,Male
24,43630,210000000,3,459359555,X-Men: The Last Stand,6.3,3525,4786,2006,May,Wednesday,Brett Ratner,4786,Male
43,43660,180000000,67,748806957,"The Chronicles of Narnia: The Lion, the Witch ...",6.7,2629,4774,2005,Dec,Wednesday,Andrew Adamson,4774,Male
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1342,47172,5000000,18,26888376,Clerks II,6.9,398,5369,2006,May,Thursday,Kevin Smith,5369,Male
1359,47275,0,3,0,A Cock and Bull Story,6.2,43,5549,2005,Jul,Sunday,Michael Winterbottom,5549,Male
1365,47328,4200000,16,0,High School Musical,6.1,1000,5113,2006,Jan,Friday,Kenny Ortega,5113,Male
1373,47351,4000000,17,31584901,Scoop,6.4,420,5457,2006,Jul,Thursday,Woody Allen,5457,Male


In [ ]:
movie_praduced_more_70 = data[data['year'].isin([2005,2006])]
movie_praduced_more_70

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
6,43607,270000000,57,391081192,Superman Returns,5.4,1400,4772,2006,Jun,Wednesday,Bryan Singer,4772,Male
8,43609,200000000,145,1065659812,Pirates of the Caribbean: Dead Man's Chest,7.0,5246,4763,2006,Jun,Tuesday,Gore Verbinski,4763,Male
18,43621,207000000,61,550000000,King Kong,6.6,2337,4777,2005,Dec,Wednesday,Peter Jackson,4777,Male
24,43630,210000000,3,459359555,X-Men: The Last Stand,6.3,3525,4786,2006,May,Wednesday,Brett Ratner,4786,Male
43,43660,180000000,67,748806957,"The Chronicles of Narnia: The Lion, the Witch ...",6.7,2629,4774,2005,Dec,Wednesday,Andrew Adamson,4774,Male
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1342,47172,5000000,18,26888376,Clerks II,6.9,398,5369,2006,May,Thursday,Kevin Smith,5369,Male
1359,47275,0,3,0,A Cock and Bull Story,6.2,43,5549,2005,Jul,Sunday,Michael Winterbottom,5549,Male
1365,47328,4200000,16,0,High School Musical,6.1,1000,5113,2006,Jan,Friday,Kenny Ortega,5113,Male
1373,47351,4000000,17,31584901,Scoop,6.4,420,5457,2006,Jul,Thursday,Woody Allen,5457,Male


In [ ]:
result = data.groupby("year").filter(lambda x: x["title"].count() > 70)
result

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
6,43607,270000000,57,391081192,Superman Returns,5.4,1400,4772,2006,Jun,Wednesday,Bryan Singer,4772,Male
8,43609,200000000,145,1065659812,Pirates of the Caribbean: Dead Man's Chest,7.0,5246,4763,2006,Jun,Tuesday,Gore Verbinski,4763,Male
18,43621,207000000,61,550000000,King Kong,6.6,2337,4777,2005,Dec,Wednesday,Peter Jackson,4777,Male
24,43630,210000000,3,459359555,X-Men: The Last Stand,6.3,3525,4786,2006,May,Wednesday,Brett Ratner,4786,Male
43,43660,180000000,67,748806957,"The Chronicles of Narnia: The Lion, the Witch ...",6.7,2629,4774,2005,Dec,Wednesday,Andrew Adamson,4774,Male
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1342,47172,5000000,18,26888376,Clerks II,6.9,398,5369,2006,May,Thursday,Kevin Smith,5369,Male
1359,47275,0,3,0,A Cock and Bull Story,6.2,43,5549,2005,Jul,Sunday,Michael Winterbottom,5549,Male
1365,47328,4200000,16,0,High School Musical,6.1,1000,5113,2006,Jan,Friday,Kenny Ortega,5113,Male
1373,47351,4000000,17,31584901,Scoop,6.4,420,5457,2006,Jul,Thursday,Woody Allen,5457,Male


To address this requirement, we initially aggregated movie counts by year using a frequency distribution on the year column.
The analysis revealed that only two years recorded production volumes exceeding 70 movies.
Subsequently, we filtered the dataset for those years to retrieve detailed movie-level information for further evaluation.

# **Q8 - How many different movies have been directed by James Cameron between year 2005 to 2010?**

In [ ]:
directed_movie= data[data['director_name']=='James Cameron']
directed_movie

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday,James Cameron,4762,Male
19,43622,200000000,100,1845034188,Titanic,7.5,7562,4762,1997,Nov,Tuesday,James Cameron,4762,Male
170,43876,100000000,101,520000000,Terminator 2: Judgment Day,7.7,4185,4762,1991,Jul,Monday,James Cameron,4762,Male
173,43879,115000000,38,378882411,True Lies,6.8,1116,4762,1994,Jul,Thursday,James Cameron,4762,Male
344,44184,70000000,24,90000098,The Abyss,7.1,808,4762,1989,Aug,Wednesday,James Cameron,4762,Male
1100,46000,18500000,67,183316455,Aliens,7.7,3220,4762,1986,Jul,Friday,James Cameron,4762,Male
1320,47036,6400000,74,78371200,The Terminator,7.3,4128,4762,1984,Oct,Friday,James Cameron,4762,Male


In [ ]:
directed_movie[directed_movie['year'].between(2005, 2010)]

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday,James Cameron,4762,Male


In [ ]:
#without merge

In [ ]:
directors.head(2)

,director_name,id,gender
0,James Cameron,4762,Male
1,Gore Verbinski,4763,Male


In [ ]:
james_id= directors.loc[directors['director_name']== 'James Cameron'] ['id'] [0]
print(james_id)


4762


In [ ]:
movies.head(2)

,id,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday


In [ ]:
james_movie= movies.loc[(movies['director_id'] == james_id) & (movies['year']>=2005) & (movies['year']<=2010)] ['title'] [0]
print(james_movie)

Avatar


As per the analyzed dataset, only **one movie (Avatar**) was directed by James Cameron between the years 2005 and 2010.

In [ ]:
#making it more standard



def solve(director_name, start_year, end_year):

  df = directors.loc[directors["director_name"] == director_name]

  if len(df) != 0:

    director_id = df.reset_index(drop=True)['id'][0]

    director_movie= movies.loc[movies['director_id']== director_id]

    res= director_movie[(director_movie['year']>=start_year) & (director_movie['year']<=end_year)]

    return res

  else:

    return "Director Not Found!"

# **Q9 - In which decade, the most number of movies were produced?**

In [ ]:
data.describe()

,id_x,budget,popularity,revenue,vote_average,vote_count,director_id,year,id_y
count,1465.000000,1.465000e+03,1465.000000,1.465000e+03,1465.000000,1465.000000,1465.000000,1465.000000,1465.000000
mean,45225.191126,4.802295e+07,30.855973,1.432539e+08,6.368191,1146.396587,5040.192491,2002.615017,5040.192491
std,1189.096396,4.935541e+07,34.845214,2.064918e+08,0.818033,1578.077438,258.059631,8.680141,258.059631
min,43597.000000,0.000000e+00,0.000000,0.000000e+00,3.000000,1.000000,4762.000000,1976.000000,4762.000000
25%,44236.000000,1.400000e+07,11.000000,1.738013e+07,5.900000,216.000000,4845.000000,1998.000000,4845.000000
50%,45022.000000,3.300000e+07,23.000000,7.578164e+07,6.400000,571.000000,4964.000000,2004.000000,4964.000000
75%,45990.000000,6.600000e+07,41.000000,1.792469e+08,6.900000,1387.000000,5179.000000,2009.000000,5179.000000
max,48395.000000,3.800000e+08,724.000000,2.787965e+09,8.300000,13752.000000,6204.000000,2016.000000,6204.000000


In [ ]:
year_bin = [1970, 1980, 1990, 2000, 2010, 2020]
bin_labels = ["A", "B", "C", "D", "E"]

In [ ]:
data["year_bucket"] = pd.cut(data["year"], bins = year_bin, labels = bin_labels)
data.head()

,id_x,budget,popularity,revenue,title,vote_average,vote_count,director_id,year,month,day,director_name,id_y,gender,year_bucket
0,43597,237000000,150,2787965087,Avatar,7.2,11800,4762,2009,Dec,Thursday,James Cameron,4762,Male,D
1,43598,300000000,139,961000000,Pirates of the Caribbean: At World's End,6.9,4500,4763,2007,May,Saturday,Gore Verbinski,4763,Male,D
2,43599,245000000,107,880674609,Spectre,6.3,4466,4764,2015,Oct,Monday,Sam Mendes,4764,Male,E
3,43600,250000000,112,1084939099,The Dark Knight Rises,7.6,9106,4765,2012,Jul,Monday,Christopher Nolan,4765,Male,E
4,43602,258000000,115,890871626,Spider-Man 3,5.9,3576,4767,2007,May,Tuesday,Sam Raimi,4767,Male,D


In [ ]:
data["year_bucket"].value_counts()

,count
year_bucket,
D,644
C,376
E,293
B,126
A,26


I performed analytics on the movie dataset by creating year-based bins to represent different decades and assigned appropriate labels.
I then calculated the frequency distribution using the value_counts() function to identify which decade had the highest number of movies produced.

Our analysis indicates that the **2000–2010 decade** experienced the highest volume of movie production. This suggests a significant expansion phase in the film industry during that period.

# **Q10 - Within the decade with most number of movies, which were the top 5 contributing directors (who produced most)?**

In [ ]:
data_D = data[data['year_bucket']=='D']
len(data_D)

644

In [ ]:
data_D['director_name'].value_counts().head(5)

,count
director_name,
Clint Eastwood,9
Robert Rodriguez,9
Woody Allen,8
Shawn Levy,7
Adam Shankman,7


Within the highest-producing decade, the top contributing directors were Clint Eastwood and Robert Rodriguez, each directing 9 movies. They were followed by Woody Allen with 8 movies, and Shawn Levy and Adam Shankman with 7 movies each.